python/tvm/error.py

```diff
@@ -0,0 +1,96 @@
+"""Structured error classes in TVM.
+
+Each error class takes an error message as its input.
+See the example sections for for suggested message conventions.
+To make the code more readable, we recommended developers to
+copy the examples and raise errors with the same message convention.
+"""
+from ._ffi.base import register_error, TVMError
+
+@register_error
+class InternalError(TVMError):
+    """Internal error in the system.
+
+    Examples
+    --------
+    .. code :: c++
+
+        // Example code C++
+        LOG(FATAL) << "InternalError: internal error detail.";
+
+    .. code :: python
+
+        # Example code in python
+        raise InternalError("internal error detail")
+    """
+    def __init__(self, msg):
+        # Patch up additional hint message.
+        if "TVM hint:" not in msg:
+            msg += ("\nTVM hint: You hit an internal error. " +
+                    "Please open a thread on https://discuss.tvm.ai/ to report it.")
+        super(InternalError, self).__init__(msg)
+
+
+register_error("ValueError", ValueError)
+register_error("TypeError", TypeError)
+
+
+@register_error
+class OpError(TVMError):
+    """Base class of all operator errors in fronends."""
+
+
+@register_error
+class OpNotImplemented(OpError):
```

Comment:
Nice, I like how the Python and C++ error handling is unified. Should `OpNotImplemented` also inherit `NotImplementedError`? In case one is trying to catch specific error types, e.g., to gather a list of all operators or attributes that are unsupported.